mmv1/products/securesourcemanager/BranchRule.yaml

```diff
@@ -0,0 +1,139 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'BranchRule'
+base_url: 'projects/{{project}}/locations/{{location}}/repositories/{{repository_id}}/branchRules?branch_rule_id={{branch_rule_id}}'
+self_link: 'projects/{{project}}/locations/{{location}}/repositories/{{repository_id}}/branchRules/{{branch_rule_id}}'
+immutable: true
```

Comment:
For branch rule, I don't think it should be immutable

Reply:
That's a good point. During manual testing, terraform did replace branch rule fields I wanted to update rather than destroy and re-create the entire branch rule. So it didn't seem to follow an immutable pattern, but it makes sense to remove this tag.

Reply:
Removed immutable.